### PR TITLE
CC-277-Implement-Additional-Meal-Plan-API

### PR DESCRIPTION
**Pull request by Binh-SPU**
# Features:
1. Implement delete Meal Plan API and edit Meal Plan API.
2. Fixing the bookmark feature in Community Page.
# Patches:
1. Fix an issue where we have just saved an meal plan, the grocery list does not update with most recent grocery list. It will now. Also, it you are in the grocery list tab, it will switch to mini meal plan tab and update the grocery list still. **Noticeably, the grocery list will add ingredients together if it duplicated. For example, 2 pancake with 20g sugar => 1 pancake with 40g sugar in grocery list. The grocery list should be updated whenever you guys add or delete anything whether in edit mode or creating mode.** Please test this extensively.
3. Fix an issue where duplicate items will show warnings. It won't now.
4. Fix an issue where we cannot edit the meal plan period before. We now can. You can add recipes, changes dates, and delete without problems (hopefully...). **Also, it is a deliberate decision to make 'after saving changes', it jumps back to main meal planner page. It is more simpler to code that way.**
5. Fix an issue where if we delete an items that have duplicates, it will delete all. It now will only delete that specific item from all of duplicates items.
6. Fix an issue where the undo button does not immediately reflect the back to the previous stage. It will undo everything including the meal plan period time.

# Known Bugs:
1) When mounting the page:
image
=> I won't do anything about this.

### CODE REVIEW

File: src/Pages/CommunityPage.js [hunk 1]
```diff
@@ -110,23 +110,25 @@ export default function CommunityPage({ navigation }) {
...
+            const isRecipeExist = recipes.some((recipeCollection) => {
+              return recipeCollection.recipes.some((info) => {
+                return info.id === item.recipe.recipeId;
+              });
+            });
             return isRecipeExist;
           })
         );
+
+        // console.log("Bookmark:", bookmark);
```
Comment (lethienantran) [inline src/Pages/CommunityPage.js:122]: please remove this line

**Comment by lethienantran** [comment]
Conflict

**Comment by Binh-SPU** [comment]
I have fixed the conflicts

File: src/Pages/CommunityPage.js [hunk 2]
```diff
@@ -198,7 +198,17 @@ export default function CommunityPage({ navigation }) {
...
       const handleRecipeToCollection = async () => {
-        await AddRecipeToCollection(recipeId);
+        const responseObject = await GetRecipeFromCollection(
+          selectedCollectionId
+        );
+        const isRecipeExist = responseObject.recipes.some(
+          (recipe) => recipe.id === post[popUpIndex].recipe.recipeId
+        );
+
+        !isRecipeExist
```
Comment (lethienantran) [inline src/Pages/CommunityPage.js:208]: can you revert this back to dev code? we just confirm in scrumptious group chat.

**Pull request merged**